org.civicrm.reqangular/src/common/services/api.js

```diff
@@ -42,6 +42,20 @@ define([
       return JSON.stringify(_.merge(defaults, params || {}));
     }
 
+    /**
+     * Constructs a GET query string exactly the same way as $http service does
+     *
+     * @param  {Object} json that is passed as "params" to $http
+     * @return {String} query string
+     */
+    function jsonToQueryString (json) {
+      return '?' + Object.keys(json).sort().map(function (key) {
```

Comment:
Do you think you could use jQuery's param method here?
http://api.jquery.com/jquery.param/

```js
$.param({a: 1, b: 2, c: "100"}); // a=1&b=2&c=100
$.param({a: [1, 2, 3]}); // a%5B%5D=1&a%5B%5D=2&a%5B%5D=3
```

Reply:
I was thinking about this, but there are 2 reasons for not doing this:
1) jQuery is a global thing
2) jQuery.param would not sort query params, while $http does, we would need to also use smth like `_(json).toPairs().sortBy(0).fromPairs().value()`.

I would honestly leave this function as it is now.

Reply:
Doesn't [`$httpParamSerializer`](https://docs.angularjs.org/api/ng/service/$httpParamSerializer) do the same